controller/src/lib.rs

```diff
@@ -1,4 +1,4 @@
-mod controller;
+pub mod controller;
```

Comment:
I don't find `pub` is necessary here, any reason this has to be `pub`?

Reply:
Since we make create multiple error classifications, I have an error module `controllerclient_error` under controller.rs. Then my new changes on `mian.rs` need use `controllerclient_error` error type as resource, so I have to make it pub.

```
pub mod controller_error {
    use controller::controller::controllerclient_error;
    use snafu::Snafu;
```

Reply:
Can we do the following in `lib.rs`
```
pub use crate::controller::BrupopController;
```
And do in `main.rs`?
```
use controller::controllerclient_error;
```